src/bitmessagemain.py

```diff
@@ -156,12 +157,9 @@ def start(self):
 
         set_thread_name("PyBitmessage")
 
-        state.dandelion_enabled = config.safeGetInt('network', 'dandelion')
-        # dandelion requires outbound connections, without them,
-        # stem objects will get stuck forever
-        if state.dandelion_enabled and not config.safeGetBoolean(
-                'bitmessagesettings', 'sendoutgoingconnections'):
-            state.dandelion_enabled = 0
+        # dandelion enabled checks at startup
+        network.dandelion.instance.is_dandelion_enabled(config)
```

Comment:
how about making this into a static method that returns `True` / `False`? Then the instance and the config are decoupled, you pass the result of `is_dandelion_enabled` do the constructor.

Reply:
but the problem is we need a variable where we can store the return value. currently after checking we are storing in class attr.

Reply:
Hmm, now I see the problem. We then probably could create a `Placeholder` instance inside `dandelion.py` and create a proper instance here in `bitmessagemain.py`. Then we can pass the value to the constructor.

Reply:
invthread.py", line 55, in run
handleExpiredDandelion(dandelion.instance.expire())

- ivthread initialized before dandelion instance created and checking for dandelion expire. real dandelion instance is required at the starting point of thread.

Reply:
Resolved now. moved Dandelion instance  into `network/__init__.py`